### ISSUE TITLE
Hide files/folders from file explorer

### ISSUE THREAD

**Issue by vigo** [zed-industries/zed#15761]
### Check for existing issues

- [X] Completed

### Describe the feature

I'm not sure if this is a bug or feature, is there anyway to hide `.ruff_cache`, `__pycache__` kind of folders from file explorer?

### If applicable, add mockups / screenshots to help present your vision of the feature

<img width="332" alt="image" src="https://github.com/user-attachments/assets/cfab4657-34cc-487d-9825-41b46e2ee91f">

**Comment by failable** [comment]
Respecting `.gitignore` could also be an sane default.

**Comment by notpeter** [comment]
I think `file_scan_exclusions` in settings is what you're looking for. Currently there is no web documentation, it's only documented in the default settings file (`cmd-shift-p`, `default settings`)

If you would like to have Zed entirely hide other folders you'll need to replace the default for `file_scan_exclusions` in your settings.json:

```json
{
  "file_scan_exclusions": [
    "**/.ruff_cache",
    // default values below
    "**/.git",
    "**/.svn",
    "**/.hg",
    "**/CVS",
    "**/.DS_Store",
    "**/Thumbs.db",
    "**/.classpath",
    "**/.settings"
  ]
}
```

**Comment by daveycodez** [comment]
Can't hide my gitignore files.. uninstalling :/

**Comment by notpeter** [comment]
> Can't hide my gitignore files.. uninstalling :/

This just merged to main 14hrs after you commented:
- https://github.com/zed-industries/zed/pull/26636

It just missed the release window, so it'll be available in Zed Preview v0.181.0 next Wednesday 2025-04-03.

**Comment by l1x** [comment]
> Can't hide my gitignore files.. uninstalling :/

Easy come, easy go.
